arcgis-ios-sdk-samples/Display information/Create symbol styles from web styles/CreateSymbolStylesFromWebStylesLegendTableViewController.swift

```diff
@@ -0,0 +1,47 @@
+// Copyright 2021 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class CreateSymbolStylesFromWebStylesLegendTableViewController: UITableViewController {
+    /// The symbols to display in the legend table.
+    var symbols = [(name: String, symbol: AGSSymbol)]()
+    
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        1
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        symbols.count
+    }
+    
+    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        "Symbol Styles"
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "SymbolLegendCell", for: indexPath)
+        // Display the name of the symbol.
+        cell.textLabel?.text = symbols[indexPath.row].name
+        // Create an icon swatch for the symbol and update the cell.
+        symbols[indexPath.row].symbol.createSwatch { (image, _) in
+            if let updateCell = tableView.cellForRow(at: indexPath) {
+                updateCell.imageView?.image = image
+                updateCell.setNeedsLayout()
+            }
+        }
```

Comment:
The reason why I have them here is I copied from https://github.com/Esri/arcgis-runtime-samples-ios/blob/d58660d404c3bc757f4d3e0a235721d5d8e41211/arcgis-ios-sdk-samples/Display%20information/Show%20legend/MILLegendTableViewController.swift#L77-L82

because I tend to keep unrelated UI code consistent among samples. But in this particular case it doesn't seem to be a good practice.

I'll take your suggestions and re-implemented this section of code. ✅

Reply:
[`ReadSymbolsFromMobileStyleSymbolSettingsViewController`](https://github.com/Esri/arcgis-runtime-samples-ios/blob/d58660d404c3bc757f4d3e0a235721d5d8e41211/arcgis-ios-sdk-samples/Display%20information/Read%20symbols%20from%20a%20mobile%20style/ReadSymbolsFromMobileStyleSymbolSettingsViewController.swift#L183-L185) would make a much better reference for how to do asynchronous image loading for table view cells.

Reply:
Thanks for the suggestions and the example. I've extracted these methods into a data source class, so that

- Even if the `AGSSymbol`s get updated, the legends still reload/display correctly
- Get rid of the additional custom `Create...LegendTableViewController` class, since all we need is a plain table
- Load image swatches in the correct place and allow cancellation

I'm not sure if I'm making things too complex here though. Since there are quite a few changes, no hurry to review!

Reply:
> Another good practice is to handle changes to the backing store. Right now, if symbols were to change after the table is loaded, nothing would happen. Adding a property observer that reloads the table view, if the view has already loaded, solves that problem.

Hi Phil, I'm not sure if I'm understanding correctly here. 😕 Could you explain a bit more?

Do you mean to pass the `symbols` as an `inout` parameter to the data source class's initializer, then observe the value; or add a property observer to the `symbols` in the view controller, and once it get changed, notify the data source class to reload the tableview?

Thanks.

Reply:
No, I meant using a property observer, like you've done.